CONTRIBUTING.md

```diff
@@ -0,0 +1,216 @@
+* Discuss a high-level topic or idea (for example, community, vision or policies)
+* Propose a new feature or other project idea
+
+Tips for communicating on issues:
+
+* **If you see an open issue that you want to tackle**, comment on the issue to let people know you’re on it. That way, people are less likely to duplicate your work.
+* **If an issue was opened a while ago**, it’s possible that it’s being addressed somewhere else, or has already been resolved, so comment to ask for confirmation before starting work.
+* **If you opened an issue, but figured out the answer later on your own**, comment on the issue to let people know, then close the issue. Even documenting that outcome is a contribution to the project.
+
+> **Note:** See in [GitHub Help](https://help.github.com/en) how to [creating an issue](https://help.github.com/en/articles/creating-an-issue).
+
+## Opening a pull request
+
+You should usually open a pull request in the following situations:
+
+* Submit trivial fixes (for example, a typo, a broken link or an obvious error).
+* Start work on a contribution that was already asked for, or that you’ve already discussed, in an issue.
+
+A pull request doesn’t have to represent finished work. It’s usually better to open a pull request early on, so others can watch or give feedback on your progress. Just mark it as a “WIP” (Work in Progress) in the subject line. You can always add more commits later.
+
+Always reference the issue related to the PR with `#number-of-issue` and use the markdown to make your comments more readable.
+
+> **Note:** If this is your first pull request, check out [Make a Pull Request](http://makeapullrequest.com/), which [**@kentcdodds**](https://github.com/kentcdodds) created as a walkthrough video tutorial. You can also practice making a pull request in the [First Contributions](https://github.com/Roshanjossey/first-contributions) repository, created by [**@Roshanjossey**](https://github.com/Roshanjossey).
+
+## Coding conventions
+
+**Adhere to standard Python style guidelines**
+All contributions to Netpyne are checked against style guidelines described in PEP 8. We also check for common coding errors (such as variables that are defined but never used).
+
+Several text editors or IDEs also have Python style checking, which can highlight style errors while you code (and train you to make those errors less frequently). This functionality is built-in to the Spyder IDE, but most editors have plug-ins that provide similar functionality. Search for `python linter <name of your favorite editor>` to learn more.
+
+**Use consistent variable naming**
+Classes should be named using `CamelCase`. Functions and instances/variables should use `snake_case` (`n_samples` rather than `nsamples`). Avoid single-character variable names, unless inside a [comprehension](https://docs.python.org/3/glossary.html#term-list-comprehension) or [generator](https://docs.python.org/3/tutorial/classes.html#tut-generators).

```

Comment:
Functions and instances/variables should use `camelCase` -- [this is how it's currently written]
nSamples

Reply:
:+1: I replace for:
**Use consistent variable naming**
Functions and instances/variables should use `CamelCase` (`nSamples` rather than `n_samples`). Avoid single-character variable names, unless inside a [comprehension](https://docs.python.org/3/glossary.html#term-list-comprehension) or [generator](https://docs.python.org/3/tutorial/classes.html#tut-generators).